Blocks: Update packages, fix deprecations & new configurations

ryelle opened the pull request:
Update the `@wordpress/*` packages, which includes some configuration changes and component prop deprecations. Remove deprecated `isDefault` prop from `<Button />`. Update eslint config to use `@wordpress/eslint-plugin/recommended-with-formatting`, which uses eslint for formatting rules instead of prettier. This prevents the prettier ruleset from running before we're ready to use it. Lastly, this adds the prettier config & format command, so we can test out whether we want to adopt the auto-formatting features.

### How to test the changes in this Pull Request:

1. Run `yarn` to update local packages
2. Building, testing, etc should all work
3. Linters should pass
4. Try running `yarn workspace wordcamp-blocks run format:js` to test out the prettier commands

ryelle committed:
Blocks: Update packages

ryelle committed:
Blocks: Remove deprecated `isDefault` prop from Button

ryelle committed:
Blocks: Update eslint rule to use eslint for formatting rules

This prevents the prettier ruleset from running before we're ready to use it.

ryelle committed:
Blocks: Add prettier config & format command

coreymckrill commented:
Haven't fully tested this yet, but just wanted to note that the Button component's `isLarge` is being deprecated in WP 5.4:

https://make.wordpress.org/core/2020/02/13/an-updated-button-component-in-wordpress-5-4/

So may be worth taking that into account for this PR, since we're making other button changes here as well.

ryelle committed:
Remove unofficially deprecated `isLarge` prop from Buttons

ryelle commented:
Fixed the merge conflict, and removed `isLarge`. It's not officially deprecated, which is why I missed it — but it doesn't do anything anymore, so it should be removed.
